pkg/controller/eventinstance/eventinstance.go

```diff
@@ -0,0 +1,91 @@
+package eventinstance
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	v1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	"github.com/acorn-io/acorn/pkg/config"
+	"github.com/acorn-io/baaah/pkg/router"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const defaultTTL = 24 * time.Hour
+
+func GCExpired() router.HandlerFunc {
+	// last stores pre-parsed TTLs from the configuration.
+	var last struct {
+		sync.Mutex
+		raw string
+		ttl time.Duration
+	}
```

Comment:
The only real issue I have with this approach is that it makes the memory difficult to reason about.

For example, typically you should use a `*sync.Mutex` to ensure that things don't get copied. However, I _think_ that the closure takes the address of the `last` variable so it's OK here?

Is it worth just making `*sync.Mutex` a pointer to avoid confusion?

Reply:
Ah, great point.

`go vet` would have probably complained if there was a copy going on, and the struct is anonymous and tightly scoped to the function for that reason, but embedding a `sync.Mutex` is definitely not idiomatic.

that said, I haven't noticed `mu := new(sync.Mutex)` very often, so I'm wondering if that's also not idiomatic.

I went for a compromise and added a named field for `mu`, then made `last` a pointer.

Reply:
I've completely refactored this. No longer using mutexes directly.